oauth2/oauth2_auth_code_test.go

```diff
@@ -969,21 +969,40 @@ func TestAuthCodeWithMockStrategy(t *testing.T) {
 					})
 
 					t.Run("should not override session data if token refresh hook returns no content", func(t *testing.T) {
+						if strat.d != "jwt" {
+							t.Skip()
```

Comment:
This test would work for both JWT and opaque if you use:

```
accessTokenClaims := testhelpers.IntrospectToken(t, oauthConfig, &refreshedToken, ts)
```

https://github.com/ory/hydra/pull/3146/commits/82a919edb23de6729ceab4fc48d4471199f25bff#diff-6d883efffdabd9715dc9872121018df30a5843c81e25dc6c4af2c3edc13fb21cR957

The problem is that the JWT strategy works a bit different so we need to ensure this works as intended for both strategies :)

Reply:
Ah, that's what I was missing. Thanks! Should be updated now.